sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/BeamCalciteTable.java

```diff
@@ -61,6 +68,39 @@ public RelDataType getRowType(RelDataTypeFactory typeFactory) {
     return CalciteUtils.toCalciteRowType(this.beamTable.getSchema(), typeFactory);
   }
 
+  @Override
+  public Statistic getStatistic() {
+    return new Statistic() {
+      @Override
+      public Double getRowCount() {
+        Double rc =
+            beamTable.getRowCount(BeamEnumerableConverter.createPipelineOptions(pipelineOptions));
+        rc = (rc == null) ? Statistics.UNKNOWN.getRowCount() : rc;
```

Comment:
If you had a custom statistic class, this would look something like:
```
return statistic.isUnknown()
  ? Statistics.UNKNOWN
  : Statistics.of(statistic.getRowCount(), ImmutabelList.of());
```